src/PluginsRegistry.js

```diff
@@ -69,7 +69,7 @@ export default class PluginsRegistry {
   /**
    * Call `onUpdate` hooks.
    */
-  onUpdate(data: Object, rule: Rule, sheet: StyleSheet): void {
+  onUpdate(data: Object | void, rule: Rule, sheet: StyleSheet): void {
```

Comment:
I am wondering if its a good thing to have data optional, why would someone call an update without data?

Reply:
Yep. These tests are failed if we do not allow to pass `undefined`.
```
FAILED TESTS:
  jss-default-unit
    support function values
      ‚úñ should add default unit
        Chrome 66.0.3359 (Mac OS X 10.12.6)
      Error: expected '' to equal '.a-id {\n  width: 1px;\n}'
          at Assertion.assert (webpack:///~/expect.js/index.js:96:0 <- tests/index.js:8804:13)
          at Assertion.be.Assertion.equal (webpack:///~/expect.js/index.js:216:0 <- tests/index.js:8924:10)
          at Assertion.(anonymous function) [as be] (webpack:///~/expect.js/index.js:69:0 <- tests/index.js:8777:24)
          at Context.<anonymous> (webpack:///~/jss-default-unit/tests/index.test.js:352:0 <- tests/index.js:20470:53)

  jss-nested
    function values
      ‚úñ should generate correct CSS
        Chrome 66.0.3359 (Mac OS X 10.12.6)
      Error: expected '' to equal '.a-id:hover {\n  color: red;\n}'
          at Assertion.assert (webpack:///~/expect.js/index.js:96:0 <- tests/index.js:8804:13)
          at Assertion.be.Assertion.equal (webpack:///~/expect.js/index.js:216:0 <- tests/index.js:8924:10)
          at Assertion.(anonymous function) [as be] (webpack:///~/expect.js/index.js:69:0 <- tests/index.js:8777:24)
          at Context.<anonymous> (webpack:///~/jss-nested/tests/index.test.js:470:0 <- tests/index.js:22087:45)

  jss-camel-case
    function values
      ‚úñ should generate correct CSS
        Chrome 66.0.3359 (Mac OS X 10.12.6)
      Error: expected '' to equal '.a-id {\n  font-size: 12;\n}'
          at Assertion.assert (webpack:///~/expect.js/index.js:96:0 <- tests/index.js:8804:13)
          at Assertion.be.Assertion.equal (webpack:///~/expect.js/index.js:216:0 <- tests/index.js:8924:10)
          at Assertion.(anonymous function) [as be] (webpack:///~/expect.js/index.js:69:0 <- tests/index.js:8777:24)
          at Context.<anonymous> (webpack:///~/jss-camel-case/tests/index.test.js:144:0 <- tests/index.js:19716:53)

  jss-vendor-prefixer
    prefix function values
      ‚úñ should generate correct CSS
        Chrome 66.0.3359 (Mac OS X 10.12.6)
      Error: expected '' to equal '.a-id {\n  display: flex;\n}'
          at Assertion.assert (webpack:///~/expect.js/index.js:96:0 <- tests/index.js:8804:13)
          at Assertion.be.Assertion.equal (webpack:///~/expect.js/index.js:216:0 <- tests/index.js:8924:10)
          at Assertion.(anonymous function) [as be] (webpack:///~/expect.js/index.js:69:0 <- tests/index.js:8777:24)
          at Context.<anonymous> (webpack:///~/jss-vendor-prefixer/tests/index.test.js:153:0 <- tests/index.js:22568:53)

  Integration: rules
    rule.toJSON()
      ‚úñ should handle function values
        Chrome 66.0.3359 (Mac OS X 10.12.6)
      Error: expected {} to sort of equal { color: 'red' }
          at Assertion.assert (webpack:///~/expect.js/index.js:96:0 <- tests/index.js:8804:13)
          at Assertion.eql (webpack:///~/expect.js/index.js:230:0 <- tests/index.js:8938:10)
          at Context.<anonymous> (webpack:///tests/integration/rules.js:438:45 <- tests/index.js:16268:56)
```